src/main/java/io/reactivex/plugins/RxJavaPlugins.java

```diff
@@ -929,6 +967,56 @@ public static Completable onAssembly(Completable source) {
     }
 
     /**
+     * Called before an operator attempts a blocking operation
+     * such as awaiting a condition or signal
+     * and should return true to indicate the operator
+     * should not block but throw an IllegalArgumentException.
+     * @return true if the blocking should be prevented
+     * @see #setFailOnNonBlockingScheduler(boolean)
+     * @since 2.0.5 - experimental
+     */
+    @Experimental
+    public static boolean onBeforeBlocking() {
+        BooleanSupplier f = onBeforeBlocking;
+        if (f != null) {
+            try {
+                return f.getAsBoolean();
+            } catch (Throwable ex) {
+                throw ExceptionHelper.wrapOrThrow(ex);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Set the handler that is called when an operator attemts a blocking
+     * await; the handler should return true to prevent the blocking
+     * and to signal an IllegalStateException instead.
+     * @param handler the handler to set, null resets to the default handler
+     * that always returns false
+     * @see #onBeforeBlocking()
+     * @since 2.0.5 - experimental
+     */
+    @Experimental
+    public static void setOnBeforeBlocking(BooleanSupplier handler) {
```

Comment:
Do you have a use case in mind for exposing this API in addition to just the boolean?